catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternFormulaParser.java

```diff
@@ -298,6 +306,49 @@ private FormulaElement collision() throws InternFormulaParserException {
 		return lookTree;
 	}
 
+	private FormulaElement distanceToTouchPosition() throws InternFormulaParserException {
+		String currentSpriteName = ProjectManager.getInstance().getCurrentSprite().getName();
+		int spriteCount = 0;
+
+		for (Sprite sprite : ProjectManager.getInstance().getSceneToPlay().getSpriteList()) {
+			if (sprite.getName().compareTo(currentSpriteName) == 0) {
+				spriteCount++;
+			}
+		}
+
+		if (!(spriteCount == 1)) {
+			throw new InternFormulaParserException("Parse Error, Sprite was not found");
+		}
```

Comment:
These methods in `Scene` stop, when the first sprite is found, but in this case the currentSprite and the secondSprite have to be found